pkg/pillar/types/sriovtypes.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+)
+
+const (
+	NicLinuxPath     = "/sys/class/net/"
+	NumvfsDevicePath = "/device/sriov_numvfs"
+	VfCountFieldName = "sriov-vf-count"
+)
+
+// Must match fields of EthVF in devcommon.proto
+type EthVF struct {
+	Index   uint8
+	PciLong string // BFD notation
+	Mac     string
+	VlanId  uint16
+}
+
+// VFList is list of VF for given PF (Eth device)
+type VFList struct {
+	Count uint8
+	Data  []EthVF
+}
+
+func (vfl *VFList) GetInfo(idx uint8) *EthVF {
+	for _, el := range vfl.Data {
+		if el.Index == idx {
+			return &el
+		}
+	}
+	return nil
+}
+
+// GetVf retrieve information about VFs for NIC given
+func GetVf(device string) (*VFList, error) {
```

Comment:
I propose to move this function from types for example to pkg/pillar/cmd/domainmgr/*.go where it is used

Reply:
We can do this, but why do you think we should not expose this into types package? That way it's easier to find helper functions working with VFs inside EVE code, or am I missing something?

Reply:
Well, the point is that you only use this feature in one specific package "domainmgr". In which there are no problems with the find.
In addition, in types, we store types and not functionality.

Reply:
If, after all, the section is critical, then it may make sense to transfer this implementation to another function, where you can return an error and process it correctly.